core/services/ocrcommon/transmitter.go

```diff
@@ -59,15 +82,13 @@ func (t *transmitter) FromAddress() common.Address {
 }
 
 func (t *transmitter) FromAddressForTransaction() common.Address {
-	// Use Round-Robin to select the next fromAddress.
-	nextFromAddress := t.fromAddresses[t.nextFromAddressIndex]
-
-	// Only apply round-robin logic for multiple sending keys.
-	if len(t.fromAddresses) > 1 {
-		t.nextFromAddressIndex++
-		if t.nextFromAddressIndex >= len(t.fromAddresses) {
-			t.nextFromAddressIndex = 0
-		}
+	// Default to first sending key.
+	nextFromAddress := t.fromAddresses[0]
```

Comment:
I don't think this is necessary is it? `GetRoundRobinAddress` should just return this address if it's the only one in the whitelist that is enabled. Otherwise, this key could be disabled, and we'd still use it.